pkg/registry/generic/etcd/etcd.go

```diff
@@ -440,6 +441,33 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	return e.finalizeDelete(out, true)
 }
 
+func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, listOptions *unversioned.ListOptions) (runtime.Object, error) {
```

Comment:
@wojtek-t - does the etcd v3 api remove the concept of a directory node?  if we only intend to use this from the context of namespace deletion (where we know creates are blocked), then it would be good if after all the delete nodes, we delete the immediate parent directory node in etc.